src/storage/src/source/postgres/replication.rs

```diff
@@ -216,14 +216,12 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
 
             tracing::info!(%id, "ensuring replication slot {slot} exists");
             super::ensure_replication_slot(&replication_client, slot).await?;
-            let slot_lsn = super::fetch_slot_resume_lsn(&replication_client, slot).await?;
 
             let resume_upper = Antichain::from_iter(
                 subsource_resume_uppers
                     .values()
                     .flat_map(|f| f.elements())
-                    // Advance any upper as far as the slot_lsn.
-                    .map(|t| std::cmp::max(*t, slot_lsn)),
+                    .cloned(),
```

Comment:
how was this `fetch_slot_resume_lsn` code originally justified?

Reply:
When adding a subsource, we needed to move the frontier from `0` to whatever the `slot_lsn` is (i.e. it would be at least this value because of snapshotting).

Reply:
Now that this is removed, will we still correctly hit the case that throws the `SlotCompactedPastResumePoint` below if the slot is in-fact compacted beyond our resume point?

Reply:
> When adding a subsource, we needed to move the frontier from 0 to whatever the slot_lsn is (i.e. it would be at least this value because of snapshotting).

I reached the same conclusion yesterday but I don't understand how all tests are passing now that I allow the resume lsn to regress

Reply:
I re-added this piece of code in this PR since it's unclear of what is going on here